backend/generating/SupervisionTemplate.java

```diff
@@ -0,0 +1,69 @@
+package generation;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.*;
+
+public class SupervisionTemplate{
+
+    private SupervisionInfo supervisionInfo;
+    private SupervisionQuestions questions;
+
+
+    public SupervisionTemplate(SupervisionInfo supervisionInfo, SupervisionQuestions questions) {
+        //these are the things that the user has to specify when modifying the template
+        this.supervisionInfo = supervisionInfo;
+        this.questions = questions;
+    }
+
+    public void setSupervisionHeader(){

```

Comment:
Seems like a confusing public API. Do you require users to call this before they call create for the first time?

Reply:
I've removed the create() method altogether, so I don't think we have this problem anymore.